kernelci/build.py

```diff
@@ -1385,6 +1386,60 @@ def install(self, verbose=False):
         return super().install(verbose)
 
 
+class MakePerfTool(Step):
+
+    @property
+    def name(self):
+        return 'perf'
+
+    def is_enabled(self):
+        """Check whether the perf is enabled
+
+        Return True if perf config fragment is enabled in the build
+        meta-data and CONFIG_PERF_EVENTS is in kernel config, or False
+        otherwise.
+        """
+        return 'perf' in self._meta.get(
+            'bmeta', 'kernel', 'defconfig_extras'
+        ) and self._kernel_config_enabled('PERF_EVENTS')
```

Comment:
```suggestion
        return self._kernel_config_enabled('PERF_EVENTS')
```
Please don't rely on the defconfig_extras, I can see how that would work but then the logic is not consistent with `PERF_EVENTS` being enabled in the `.config`.  So I think it's better to just rely on the `.config` and make sure the `PERF_EVENTS` is there when we want to build `perf`.

Reply:
In the latest version, I add a build_perf parameter into build_environments>arch_params. So that we can enable and disable by different arch+cc combination